File: src/main/java/org/example/Java/Act8.java
package org.example.Java;

import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class Act8 {
    Scanner scanner = new Scanner(System.in);
    public Map<String, Integer> ContarVocal() {
        System.out.println("Dame una palabra ?");
        String palabra =  scanner.nextLine();
        Map<String, Integer> map = new HashMap<>();

        for (int i = 0; i < palabra.length(); i++) {

            switch (palabra.charAt(i)) {
                case 'a':
                    map.compute("a", (key, val) -> val == null ? 1 : val + 1);
                    break;
                case 'e':
                    map.compute("e", (key, val) -> val == null ? 1 : val + 1);
                    break;
                case 'i':
                    map.compute("i", (key, val) -> val == null ? 1 : val + 1);
                    break;
                case 'o':
                    map.compute("o", (key, val) -> val == null ? 1 : val + 1);
                    break;
                case 'u':
                    map.compute("u", (key, val) -> val == null ? 1 : val + 1);
                    break;
                    default:
            }
        }
        return map;
    }
}

File: src/main/java/org/example/Kotlin/Act8.kt
package org.example.Kotlin

import java.util.HashMap

class Act8 {
    fun ContarVocal(): Map<String, Int> {
        println("Dame una palabra ?")
        val palabra: String = readLine().toString()
        val map: MutableMap<String, Int> = HashMap()

        for (i in 0 until palabra.length) {
            when (palabra[i]) {
                'a' -> map.compute("a") { key: String?, `val`: Int? -> if (`val` == null) 1 else `val` + 1 }
                'e' -> map.compute("e") { key: String?, `val`: Int? -> if (`val` == null) 1 else `val` + 1 }
                'i' -> map.compute("i") { key: String?, `val`: Int? -> if (`val` == null) 1 else `val` + 1 }
                'o' -> map.compute("o") { key: String?, `val`: Int? -> if (`val` == null) 1 else `val` + 1 }
                'u' -> map.compute("u") { key: String?, `val`: Int? -> if (`val` == null) 1 else `val` + 1 }
                else -> {}
            }
        }
        return map
    }
}